crates/consensus/src/transaction/eip4844/builder.rs

```diff
@@ -339,12 +339,13 @@ impl<T: SidecarCoder> SidecarBuilder<T> {
         self.coder.code(&mut self.inner, data);
     }
 
+    /// Build the sidecar from the data with the provided settings.
     #[cfg(feature = "kzg")]
-    /// Build the sidecar from the data.
-    pub fn build(
+    pub fn build_with_settings(
         self,
-        settings: &KzgSettings,
+        settings: &crate::EnvKzgSettings,
```

Comment:
What if we kept `KzgSettings` here and called `.get()` in `build`? This way we don't force our own Env type on this function

Reply:
sure makes sense